datafusion/core/src/physical_plan/aggregates/hash.rs

```diff
@@ -200,11 +200,6 @@ fn group_aggregate_batch(
     // evaluate the grouping expressions
     let group_by_values = evaluate_group_by(group_by, &batch)?;
 
-    // evaluate the aggregation expressions.
-    // We could evaluate them after the `take`, but since we need to evaluate all
-    // of them anyways, it is more performant to do it while they are together.
-    let aggr_input_values = evaluate_many(aggregate_expressions, &batch)?;
```

Comment:
Won't this have a negative performance impact (as mentioned in the comments)?

Reply:
I’ve checked it via running benchmark. I do not see any perf issue. 

Maybe I’m wrong :( benchmark is tricky